integration-tests/hibernate-orm-panache/src/main/java/io/quarkus/it/panache/DogDto2.java

```diff
@@ -0,0 +1,25 @@
+package io.quarkus.it.panache;
+
+import io.quarkus.hibernate.orm.panache.common.NestedProjectedClass;
+import io.quarkus.runtime.annotations.RegisterForReflection;
+
+@RegisterForReflection
+public class DogDto2 {
+    public String name;
+    public PersonDto2 owner;
+
+    public DogDto2(String name, PersonDto2 owner) {
+        this.name = name;
+        this.owner = owner;
+    }
+
+    @NestedProjectedClass
+    @RegisterForReflection
```

Comment:
Is it really needed?
If yes we should register for reflection all classes annotated by `@NestedProjectedClass`